static/js/components/forms/B2BPurchaseForm_test.js

```diff
@@ -125,6 +123,26 @@ describe("B2BPurchaseForm", () => {
       )
     })
 
+    it("should validate email field", async () => {
+      const testCases = [
+        { input: "", expectedError: "Email is a required field" },
+        { input: "something", expectedError: "Invalid email" },
+        { input: "something@", expectedError: "Invalid email" },
+        { input: "@something", expectedError: "Invalid email" },
+      ]
+
+      for (const { input, expectedError } of testCases) {
+        const values = { email: input }
+
+        try {
+          await emailValidation.validate(values)
+          assert.fail("Email validation should have thrown an error")
+        } catch (error) {
+          assert.strictEqual(error.message, expectedError)
+        }
+      }
```

Comment:
Instead of this loop here, the standard way of writing parametrized-like tests in JS can be seen in our [CheckoutForm_test](https://github.com/mitodl/mitxpro/blob/master/static/js/components/forms/CheckoutForm_test.js#L69).

Reply:
Also validating email would mean that it will show an error for the current(Invalid) inputs you added and should show no error when a  correct email is given. So you can add an additional value in the test which can be a valid email address.

Reply:
fixed